Plutarch/Internal/Term.hs

```diff
@@ -718,12 +719,15 @@ compile' t =
         (True, m) -> (m, (n, term) : defs, n + 1)
         (False, m) -> (m, defs, n)
 
+      hoistedTermRaw :: HoistedTerm -> RawTerm
+      hoistedTermRaw (HoistedTerm _ t) = t
+
       toInline :: S.Set Dig
       toInline =
         S.fromList
           . fmap (\(HoistedTerm hash _) -> hash)
           . (head <$>)
-          . filter ((== 1) . length)
+          . filter (\terms -> length terms == 1 || smallEnoughToInline (hoistedTermRaw $ head terms))
```

Comment:
I see, but why make this change? Does it fix anything?

Reply:
It's slightly more efficient when constanted value is pletted. Previously, plet stays, but with this, plet of constanted value will get automatically inlined. 

It's not ideal, because it wont' inline application of inlinable constant values, but still strictly better than original one

Reply:
ex;

```hs
pcon $ PJust $ pconstant 10

--previously: (\x -> (\a b -> a x)) 10
--new: (\a b -> a 10)
```

Reply:
It inlined constants too early so they were applied to lambdas, this instead inlines inside